### DISCUSSION TITLE
LeastSquares cost function - meaning of yerror

### DISCUSSION THREAD

**Discussion by ian-barr** [scikit-hep/iminuit#934] [Q&A]
Hi, I had a question I didn't see addressed anywhere. In the LeastSquares cost function, there is an argument called yerror, which is described as:
`Estimated uncertainty of observed values. Must have same shape as y or be a scalar, which is then broadcasted to same shape as y.`
My questions are: how are these errors incorporated into the fit, and what is the effect of changing the values of yerror?

**Comment by HDembinski** [comment]
`LeastSquares.yerror` is just a property that returns the error that you provide when you initialize this class. The errors are used as a weight for the point. LeastSquares computes the sum of [(data - prediction)/error]^2. This form can be derived as a special case of the maximum-likelihood method, valid for Gaussian fluctuations around the true expectation value.
